### PR TITLE
add multimodal projector for llama4

### PR DESCRIPTION
**Pull request by ZhaoyueCheng**
# Description

This PR 
+ adds support for Multi Modal Projector used in Llama4 to project ViT embeddings to text embeddings dim
+ adds `query_pre_attn_scalar` to match with huggingface query attention norm

# Tests

added layerwise unit test for the op

```
zhaoyuec@t1v-n-9dd93d8b-w-0:/mnt/disks/persist/maxtext$ cd /mnt/disks/persist/maxtext && PYTHONPATH=/mnt/disks/persist/maxtext python3 MaxText/tests/check_llama4_layers.py
2025-05-13 21:08:35.989279: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:467] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1747170516.004717  840951 cuda_dnn.cc:8579] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1747170516.009362  840951 cuda_blas.cc:1407] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
W0000 00:00:1747170516.022818  840951 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.
W0000 00:00:1747170516.022848  840951 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.
W0000 00:00:1747170516.022855  840951 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.
W0000 00:00:1747170516.022860  840951 computation_placer.cc:177] computation placer already registered. Please check linkage and avoid linking the same target more than once.
....
----------------------------------------------------------------------
Ran 4 tests in 74.511s

OK
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/layers/llama4.py [hunk 1]
```diff
@@ -219,6 +219,39 @@ def __call__(self, encoded_patches: Array, deterministic: bool = False) -> Array
     return result
 
 
+class Llama4MultiModalProjector(nn.Module):
```
Comment (aireenmei) [inline MaxText/layers/llama4.py:222]: If it's a simple linear layer, do we need a class for it?

Comment (hengtaoguo) [reply MaxText/layers/llama4.py:222]: +1. If so, we could setup this layer just in the Llama4VisionEncoder with something like `self.multimodal_projector=linears.DenseGeneral` without using a separate class.

Comment (ZhaoyueCheng) [reply MaxText/layers/llama4.py:237]: Thanks for the review! synced offline to keep the classes seperate similar to in [code](https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/layers/models.py#L705) so that when we are fine tuning it will be easier to freeze the visionencoder and only fine tune the embedder / adapter with the main LLM model

**Review by gagika — APPROVED** [review]
